DQM/EcalCommon/interface/EcalMonitorPrescaler.h

```diff
@@ -2,26 +2,38 @@
 #define EcalMonitorPrescaler_H
 
 #include "DataFormats/EcalRawData/interface/EcalRawDataCollections.h"
-#include "FWCore/Framework/interface/EDFilter.h"
+#include "FWCore/Framework/interface/global/EDFilter.h"
 
 #include <utility>
+#include <atomic>
 
-class EcalMonitorPrescaler : public edm::EDFilter {
+namespace ecaldqm {
+  enum Prescalers { kPhysics, kCosmics, kCalibration, kLaser, kLed, kTestPulse, kPedestal, nPrescalers };
+
+  struct PrescaleCounter {
+    PrescaleCounter() {
+      for (unsigned iP(0); iP != nPrescalers; ++iP)
+        counters_[iP] = 0;
+    }
+    mutable std::atomic<unsigned> counters_[nPrescalers];
+  };
+}  // namespace ecaldqm
+
+class EcalMonitorPrescaler : public edm::global::EDFilter<edm::RunCache<ecaldqm::PrescaleCounter>> {
 public:
   EcalMonitorPrescaler(edm::ParameterSet const &);
   ~EcalMonitorPrescaler() override;
 
-  void beginRun(edm::Run const &, edm::EventSetup const &) override;
-  bool filter(edm::Event &, edm::EventSetup const &) override;
+  std::shared_ptr<ecaldqm::PrescaleCounter> globalBeginRun(edm::Run const &, edm::EventSetup const &) const override;
+  bool filter(edm::StreamID, edm::Event &, edm::EventSetup const &) const override;
+  void globalEndRun(edm::Run const &, edm::EventSetup const &) const override;
 
 private:
-  enum Prescalers { kPhysics, kCosmics, kCalibration, kLaser, kLed, kTestPulse, kPedestal, nPrescalers };
-
-  static uint32_t filterBits_[nPrescalers];
+  static uint32_t filterBits_[ecaldqm::nPrescalers];
```

Comment:
Should this be `const` (see also the static analyzer report)?

Reply:
I believe so. There's no reason for this *not* to be `static const`. I'll update the PR and the backports